src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -930,46 +657,475 @@ public void Dispose()
+            private readonly LdaTransformer _parent;
+            private readonly ColumnType[] _srcTypes;
+            private readonly int[] _srcCols;
+
+            public Mapper(LdaTransformer parent, Schema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _srcTypes = new ColumnType[_parent.ColumnPairs.Length];
+                _srcCols = new int[_parent.ColumnPairs.Length];
+
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    if (!inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out _srcCols[i]))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+
+                    var srcCol = inputSchema[_srcCols[i]];
+                    if (!srcCol.Type.IsKnownSizeVector || !(srcCol.Type.ItemType is NumberType))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+
+                    _srcTypes[i] = srcCol.Type;
+                }
+            }
+
+            public override Schema.Column[] GetOutputColumns()
+            {
+                var result = new Schema.Column[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    var info = _parent._columns[i];
+                    result[i] = new Schema.Column(_parent.ColumnPairs[i].output, new VectorType(NumberType.Float, info.NumTopic), null);
+                }
+                return result;
+            }
+
+            protected override Delegate MakeGetter(IRow input, int iinfo, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+                disposer = null;
+
+                return GetTopic(input, iinfo);
+            }
+
+            private ValueGetter<VBuffer<float>> GetTopic(IRow input, int iinfo)
+            {
+                var getSrc = RowCursorUtils.GetVecGetterAs<Double>(NumberType.R8, input, _srcCols[iinfo]);
+                var src = default(VBuffer<Double>);
+                var lda = _parent._ldas[iinfo];
+                int numBurninIter = lda.InfoEx.NumBurninIter;
+                bool reset = lda.InfoEx.ResetRandomGenerator;
+                return
+                    (ref VBuffer<float> dst) =>
+                    {
+                        // REVIEW: This will work, but there are opportunities for caching
+                        // based on input.Counter that are probably worthwhile given how long inference takes.
+                        getSrc(ref src);
+                        lda.Output(in src, ref dst, numBurninIter, reset);
+                    };
+            }
+        }
+
+        public const string LoaderSignature = "LdaTransformer";
```

Comment:
>LdaTransformer [](start = 47, length = 14)

please revert, or it will not be back-compatible with old models

Reply:
fixed. thanks for the info

---
In reply to: [233584271](https://github.com/dotnet/machinelearning/pull/1410#discussion_r233584271) [](ancestors = 233584271)